src/Command/CommandDispatcherInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace SystemCtl\Command;
+
+use SystemCtl\Exception\CommandFailedException;
+
+interface CommandDispatcherInterface
+{
+    /**
+     * Timeout after which the dispatcher failes the execution
+     *
+     * @param int $timeout
+     *
+     * @return CommandDispatcherInterface
+     */
+    public function setTimeout(int $timeout): CommandDispatcherInterface;
+
+    /**
+     * Set basic binary to dispatch
+     *
+     * @param string $binary
+     *
+     * @return CommandDispatcherInterface
+     */
+    public function setBinary(string $binary): CommandDispatcherInterface;
+
+    /**
+     * Dispatch given commands against implementers logic and creating a new command
+     * to read results
+     *
+     * @param array $commands
+     *
+     * @return CommandInterface
+     * @throws CommandFailedException
+     */
+    public function dispatch(...$commands): CommandInterface;
```

Comment:
okay. But wouldn't this cause problems, if you add from the `listUnits` command?
The argument content would be:
```php
[
   ['list-units', 'PREFIX'],
]
```
Or doesn't this cause problems with SystemCtl?